### DISCUSSION TITLE
Server side Image generation for a common seed generation for comparison OR have users upload a sample image using a "common seed prompt" for comparison.

### DISCUSSION THREAD

**Discussion by kuso-ge** [civitai/civitai#496] [Ideas]
See TLDR below. 
Okay this might not be a feasible as I thought because of the hardware requirements to pull this off and increase operating costs.  

The idea is that every model uploaded enters a queue where the model will be used by the server to generate [sample images] using a **[common prompt]**, just so users can actually see how good the base generation of the model is. (I find that sometimes the sample images provided by the uploader doesn't really reflect what the model's base capabilities are, like, the sample images are heavily edited to look good) 

For example, let's say I uploaded a [model-a].
The server would then generate images using these common base prompts:

- [masterpiece, best quality, 1girl]
- [masterpiece, best quality, realistic, 1girl]
- [masterpiece, best quality, landscape]
- [masterpiece, best quality, dog]
- [masterpiece. best quality, dog and cat]
- [masterpiece, best quality, an apple]
- etc...

with global [negative prompt] and in a 512x512 resolution (for 1.5) or 1024x1024 resolution (for 2.0+). 

with different sampling settings(the common one at least):

- [Euler A, 20 steps], [Euler A, 24], [Euler A, 30], [Euler A, 50]
- [DPM2, 20 steps], [DPM2, 24], [DPM2, 30], [DPM2, 50]
- [<sampler>, 20 steps], [<sampler>, 24 steps], [<sampler>, 30 steps], [<sampler>, 50 steps]
- etc...

using a common [seed] and unique [seed] for each generation. 
 
This way, every model will have sample images that can easily be compared to other models. Models that are too similar can easily be find out by checking the generated images it created on the common <seed> images. (I noticed this after trying out several models that some models actually generate very similar images with the other model slightly better)

We will also know how good the model is in generating People, objects, animals, landscapes, etc...

The uploaded model doesn't need to wait for the generation queue to finish to be available. The generation queue continues in the 
background and sample images are updated to the pages automatically for every successful generation. 

The users can then vote on the models with a rating system for different generation settings.

- Landscape 3/5
- Girl 5/5
- Boy 2/5
- Animal 3/5
- etc... 

Now this would require a lot of storage space, electricity costs for the continuous image generation, essentially making this a very expensive idea. I understand that this is too big, but I'm just putting the idea out there. I think every user would benefit in actually identifying which models, mixes, etc... are actually good, are just copies of each other that are worse or slightly better or just outright duplicates. 

### **TLDR:** 
Or the website simply asks the uploader to upload **images** using a [common base settings] for every category:

- Girl, 
- Animal, 
- Boy, 
- Landscape, 
- Object, 
- etc...

**Comment by Naualon** [comment]
I dunno about a 'server side gen' but since this is a community we could release a 'guideline' for example pics so that people would add, besides their 'best in class' gens, also the default ones. That said, it would be lovely if example pics would leave loras aside, because some are adding loras to their showcase which is a step in the wrong direction, comparison wise.

**Reply by kuso-ge** [reply]
yeah, the server side generation suggestion was an ambitious idea I just put forward. But my goal is indeed to have a guideline for sample generated images so we can actually see what is the model capable off in generating other stuff like (landscape, animals, objects, buildings, crowds, different ages of man/woman, etc...) and also see which model mixes are largely the same if the same seeds and prompt are used.
